### ISSUE TITLE
Can not "Associate work items" after updating to VScode 1.16

### ISSUE THREAD

**Issue by Svachic** [Microsoft/vsts-vscode#314]
After update to VS code version 1.16, it is not possible to associate the work item with the commit.
(Using Git)

**Comment by yashv** [comment]
Noticed that it still works if the commit 'message' is in the format: "#{task_number} - {Task Heading Text}". 
Example commit message: #9999 - Fix All The Issues

**Comment by Svachic** [comment]
Thank you. I tried it and it works fine, but only when I handwrite it into the commit textbox. In previous version(1.15), if I used "Associate work item" command and click some task, it automaticly paste the hastag with the id of task into the commit textbox. Now if I do this, nothing will happen. Could it be only some settings problem?

**Comment by antoine-duc** [comment]
Hello,

Same problem here.

**Comment by EvtK** [comment]
+1

was already searching why nothing happens after selecting the work item, then stumbled upon this issue :)

**Comment by chadbr** [comment]
++++++1 (lot's on team fussing at me about it...)

**Comment by dougrday** [comment]
Any update on this?  It's definitely interfering with our ability to use this extension.

**Comment by chadbr** [comment]
fwiw,

https://github.com/Microsoft/vsts-vscode/blob/955f6b03ef41fbff3f1663466fe2c4eb9215d71b/src/team-extension.ts#L373

inputBox.value is undefined... I'm not sure if this is a regression in vscode or not.

```
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:29 [ms-vsts.team] Cannot read property 'value' of undefined: TypeError: Cannot read property 'value' of undefined
	at TeamExtension.appendToCheckinMessage (/Users/chadbr/.vscode/extensions/ms-vsts.team-1.122.0/out/src/team-extension.js:371:54)
	at TeamExtension.<anonymous> (/Users/chadbr/.vscode/extensions/ms-vsts.team-1.122.0/out/src/team-extension.js:362:26)
	at Generator.next (<anonymous>)
	at fulfilled (/Users/chadbr/.vscode/extensions/ms-vsts.team-1.122.0/out/src/team-extension.js:8:58)
	at <anonymous>```


Type def for SourceControlInputBox

```
	/**
	 * Represents the input box in the Source Control viewlet.
	 */
	export interface SourceControlInputBox {

		/**
		 * Setter and getter for the contents of the input box.
		 */
		value: string;
	}
```

**Comment by joaomoreno** [comment]
Hi guys, VS Code dev here.

There were some definite changes to the API:

```
	export namespace scm {

		/**
		 * ~~The [input box](#SourceControlInputBox) for the last source control
		 * created by the extension.~~
		 *
		 * @deprecated Use SourceControl.inputBox instead
		 */
		export const inputBox: SourceControlInputBox;
```

The [`scm.inputBox`](https://github.com/Microsoft/vscode/blob/master/src/vs/vscode.d.ts#L5769) is now deprecated. You can start using the [`SourceControl.inputBox`](https://github.com/Microsoft/vscode/blob/master/src/vs/vscode.d.ts#L5719) property instead.

The deprecated `scm.inputBox` property will stay `undefined` until an extension calls `scm.createSourceControl`, at which point it will become the `SourceControl.inputBox` of the `SourceControl` created last by your extension. The recommendation is to simply use `SourceControl.inputBox` from now on.

Sorry for the breakage!

**Comment by antoine-duc** [comment]
Thank you for this explanation but I'm just a final user of vsts-vscode extension.
Have you any idea when this will be fixed?

**Comment by chadbr** [comment]
The guy that made all the latest code changes hasn't been on github for a while so - I'm guessing it will be a while... @jeffyoung

**Comment by chadbr** [comment]
maybe @buckh can assign someone?

**Comment by buckh** [comment]
The team tells me the issue will be fixed this sprint, which means the next two weeks. Sorry this took so long.

**Comment by premiumwd** [comment]
so tomorrow this will be fixed? update @buckh

**Comment by jrbriggs** [comment]
We're going to pick this up next week.

However, if anyone wants to make a first contribution into the codebase, this is an excellent little bug to pick off and get your feet wet with! :)

**Comment by yashv** [comment]
@jrbriggs, I can take it up. However, latest npm package vscode 1.1.6 still has old definitions of "interface SourceControl". it was published 2 weeks ago, so I guess its due a refresh soon.

**Comment by chadbr** [comment]
Same... I was going to fix locally but I couldn’t figure out where to find the updated typings

**Comment by yashv** [comment]
TLDR; Seems like access to the inputBox of another source control extension ('git' in this case) is not available anymore.

I tried to workaround the old definitions problem by simply copying the latest from the repo. It helped initially, but then I realized that this extension does not 'create' a 'SourceControl' when the repository is a 'Git' repository. 
Then I noticed that VSCode has a [built-in](https://github.com/Microsoft/vscode/tree/master/extensions/git) extension for Git and that might be the default 'git - type' repository handler. 
And now that the access to scm.inputBox (last input box displayed) is taken away (I believe for keeping extensions sandboxed), this extension can access the inputBox textbox only if it creates a new SourceControl type. In that case, it will be duplicating what the built-in git extension does.

I tried creating a minimal implementation (modeling after TfvcSCMProvider), and realized that the repository appeared twice in the list. I'll upload the code to my forked repo if anyone wants to take a look.

**Comment by joaomoreno** [comment]
@yashv Read my details above:

> The deprecated `scm.inputBox` property will stay `undefined` until an extension calls `scm.createSourceControl`, at which point it will become the `SourceControl.inputBox` of the `SourceControl` created last by your extension. The recommendation is to simply use `SourceControl.inputBox` from now on.

**Comment by yashv** [comment]
@joaomoreno, I was hoping it was an easy replacement of scm.inputBox with SourceControl.inputBox. 
It seems like 'inputBox' was a global variable exposed by scm namespace.
```
export namespace scm {
			/**
			 * ~~The [input box](#SourceControlInputBox) for the last source control
			 * created by the extension.~~
			 *
			 * @deprecated Use SourceControl.inputBox instead
			 */
			export const inputBox
```
Now, its tied to the instance of a SourceControl:

		export interface SourceControl {
	
			/**
                          ........
			 */
	
			/**
			 * The [input box](#SourceControlInputBox) for this source control.
			 */
			readonly inputBox: SourceControlInputBox;

Extension doesn't create any SourceControl instances when current workspace is a Git repository. 
Is there a way to get access to existing SourceControl instances (not created by this extension) ?

**Comment by joaomoreno** [comment]
There isn't any way currently. What is the actual feature you intend to implement?

**Comment by premiumwd** [comment]
@joaomoreno associate work items for vsts in vs code extension, like in ticket bro. Really?

**Comment by yashv** [comment]
@joaomoreno,
Bigger picture: VS online detects the commit message is same as "#{VS_task_number} - {VS_Task_Heading_Text}" and associates the commit with the task.

I'm trying to set the inputBox value text (in the format mentioned above) for the currently displayed/selected SourceControl when user clicks on a Work Item among the list displayed by this extension.

**Comment by joaomoreno** [comment]
Would that make sense no matter which SCM provider is active? Git? Mercurial? Perforce?

**Comment by chadbr** [comment]
I don’t see why not...

**Comment by joaomoreno** [comment]
I've exported a thin API on top of the git extension, which lets you get a hold of the open repositories and their respective inputBoxes: https://github.com/Microsoft/vscode/issues/38389. This should be enough to close this issue.

Here's the API definition: https://github.com/Microsoft/vscode/blob/6d5112610f32429a6a6c896023af60eebee73d14/extensions/git/src/api.ts#L11:L22

Here's a sample snippet:

```ts
	const git = extensions.getExtension('vscode.git').exports;
	git.getRepositories().then(repos => {
		repos[0].inputBox.value = 'hello world'
	});
```

**Comment by dougrday** [comment]
I anyone working on this fix now that @joaomoreno has provided an API?  If not, I can jump in and PR it.

**Comment by premiumwd** [comment]
@dougrday please do, no one is working on it.

**Comment by dougrday** [comment]
@joaomoreno I think we're blocked by this issue:  https://github.com/Microsoft/vscode/issues/31103

Here's what I'm currently seeing.
```
const gitExtension = vscode.extensions.getExtension('vscode.git'); // Works
const git = gitExtension.exports; // undefined
```

**Comment by joaomoreno** [comment]
You need to test this against the latest Insider release. This change is still not in Stable.

**Comment by westerncj** [comment]
Are there any updates for this issue?

**Comment by dougrday** [comment]
I haven't had a chance to look at it again.  I'll see if I can look at it tonight.

**Comment by dougrday** [comment]
@jrbriggs @buckh Any chance that this pull request can get reviewed & merged?  I know this bug has plagued everyone for nearly 6 months now.

**Comment by kaylangan** [comment]
@dougrday we have someone on our team looking at your PR right now. Thank you for the contribution!

**Comment by dougrday** [comment]
@ermeckle Thank you for reviewing/merging!

Any word on when a new build will be pushed?  Looks like there hasn't been an update in 6 months.

**Comment by ermeckle** [comment]
I've got a couple changes of my own coming in the next week, then I'll be looking into pushing a build.

**Comment by dougrday** [comment]
Great news, thank you!

**Comment by dougrday** [comment]
@ermeckle Any update on a build?

**Comment by kaylangan** [comment]
@dougrday No, this project is not dead. In fact, we're in the process of releasing a new update imminently. :)

However, to set expectations, this project will not be updated as frequently as it was in the past. We project about 2-3 releases per year.

**Comment by dougrday** [comment]
Thank you for the update.  :)

Actually, 2-3 releases per year would be **much** better than the current schedule (it's been nearly 8 months since a release).

**Comment by visagegod** [comment]
Not being able to get the inputBox of a pre-existing source control is an absolute nightmare! Has anyone got any work arounds or is this still a major issue?

**Comment by wsyperek** [comment]
+1

**Comment by ermeckle** [comment]
This was fixed in 1.133.0

**Comment by visagegod** [comment]
> This was fixed in 1.133.0

Is there anyway to get the inputBox of a pre-existing source control?

**Comment by dougrday** [comment]
> Is there anyway to get the inputBox of a pre-existing source control?

See [src/team-extension.ts line 375](https://github.com/Microsoft/vsts-vscode/commit/b6db8eb77102f829b9945589e8759e960face353#diff-e98f1579169eb30cf7e0f2ac774d58d9R375).

**Comment by Steveja74** [comment]
Hi.  I have VSTS version 1.136.0 installed with VSCode 1.24.0.  When I select Associated with Work Item, nothing appears in the commit message...do I need to downgrade?
